### PR TITLE
yolo: if train subset already present, merge default subset into it

### PR DESCRIPTION
**Pull request by Eldies**
### Summary

YOLO formats export "default" subset and subset without name as "train".
It causes problems if "train" or "Train" subset is also present.
Fixing it by merging subsets.

### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/cvat-ai/datumaro/blob/develop/CHANGELOG.md)
- [ ] I have updated the [documentation](https://github.com/cvat-ai/datumaro/tree/develop/docs) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2022 CVAT.ai Corporation
#
# SPDX-License-Identifier: MIT
```

## Summary by CodeRabbit

- **New Features**
	- Introduced a new function for improved handling of subsets in YOLO format converters.
- **Bug Fixes**
	- Enhanced error handling and logging during the export process.
- **Tests**
	- Added a test to verify merging of default subsets into the "Train" subset during conversion.

### CODE REVIEW

File: src/datumaro/plugins/yolo_format/converter.py [hunk 1]
```diff
@@ -73,6 +73,26 @@ def rotate_point(x: float, y: float):
...
+        for subset_name, subset in initial_subsets.items()
+        if subset_name and subset_name != DEFAULT_SUBSET_NAME
+    }
+    for subset_name, subset in initial_subsets.items():
+        if not subset_name or subset_name == DEFAULT_SUBSET_NAME:
+            subset_name = YoloPath.DEFAULT_SUBSET_NAME
+            try:
+                subset_name = next(name for name in subsets if name.lower() == subset_name.lower())
+            except StopIteration:
+                pass
```
Comment (zhiltsov-max) [inline src/datumaro/plugins/yolo_format/converter.py:90]: Why not `if subset_name.lower() in subsets:`?

Comment (Eldies) [reply src/datumaro/plugins/yolo_format/converter.py:90]: subsets can have names not only with lowercase, e.g. "Train".

**Review by zhiltsov-max — APPROVED** [review]

**Review by zhiltsov-max — CHANGES_REQUESTED** [review]
Please update changelog and add description to the PR.

**Comment by Eldies** [comment]
> Please update changelog and add description to the PR.

done

**Review by zhiltsov-max — APPROVED** [review]

**Pull request merged**